Conceptpower+Spring/src/test/java/edu/asu/conceptpower/servlet/rest/ConceptLookUpTest.java

```diff
@@ -0,0 +1,149 @@
+        entry.setConceptList(Constants.WORDNET_DICTIONARY);
+        entry.setTypeId("TYPE-1");
+        entries[0] = entry;
+
+        ConceptEntry entry2 = new ConceptEntry();
+        entry2.setWord(word);
+        entry2.setPos(pos);
+        entry2.setConceptList("Test List");
+        entry2.setTypeId("TYPE-2");
+        entries[1] = entry2;
+
+        ConceptType type1 = new ConceptType();
+        type1.setTypeId(typeId1);
+        type1.setDescription("TypeID 1");
+        type1.setTypeName("TYPE-1");
+
+        ConceptType type2 = new ConceptType();
+        type2.setTypeId(typeId2);
+        type1.setDescription("TypeID 2");
+        type1.setTypeName("TYPE-2");
+
+        Mockito.when(dictManager.getConceptListEntriesForWordPOS("pony", POS.NOUN, null)).thenReturn(entries);
+        Mockito.when(typeManager.getType(typeId1)).thenReturn(type1);
+        Mockito.when(typeManager.getType(typeId2)).thenReturn(type2);
+        Mockito.when(messageFactory.getMessageFactory(MediaType.APPLICATION_XML_VALUE)).thenReturn(xmlMEssageFactory);
+        Mockito.when(messageFactory.getMessageFactory(MediaType.APPLICATION_JSON_VALUE)).thenReturn(jsonMessageFactory);
+
+        Mockito.when(xmlMEssageFactory.createConceptMessage()).thenReturn(new XMLConceptMessage(uriCreator));
+        Mockito.when(jsonMessageFactory.createConceptMessage()).thenReturn(new JsonConceptMessage(uriCreator));
+    }
+
+    @Test
+    public void getWordNetEntryTest()
+            throws IndexerRunningException, ParserConfigurationException, SAXException, IOException {
+        ResponseEntity<String> response = conceptLookup.getWordNetEntry("pony", POS.NOUN,
+                MediaType.APPLICATION_XML_VALUE);
+        RestTestUtility.testValidXml(response.getBody());
+        Assert.assertEquals(HttpStatus.OK, response.getStatusCode());
+    }
+
+    @Test
+    public void getWordNetEntryTestInJson()
+            throws IndexerRunningException, ParserConfigurationException, SAXException, IOException, JSONException {
+
+        ResponseEntity<String> response = conceptLookup.getWordNetEntry("pony", POS.NOUN,
+                MediaType.APPLICATION_JSON_VALUE);
+        RestTestUtility.testValidJson(response.getBody());
+        Assert.assertEquals(HttpStatus.OK, response.getStatusCode());
+    }
+
+    @Test
+    public void getWordNetEntryInRDF() throws IndexerRunningException, IllegalAccessException, LuceneException {
+
+        ConceptEntry[] entries = new ConceptEntry[2];
+        ConceptEntry entry1 = new ConceptEntry();
+        entry1.setWord("robert");
+        entry1.setPos(POS.NOUN);
+        entry1.setId("WID-123");
+
+        ConceptEntry entry2 = new ConceptEntry();
+        entry2.setId("WID-456");
+        entry2.setWord("robert-1");
+        entry1.setPos(POS.NOUN);
+
+        entries[0] = entry1;
+        entries[1] = entry2;
+
+        Mockito.when(dictManager.getConceptListEntriesForWordPOS("robert", POS.NOUN, null)).thenReturn(entries);
```

Comment:
these strings should be in constants or declared at the beginning of the method to make maintenance easier.